### DISCUSSION TITLE
Using dot in sentry tag

### DISCUSSION THREAD

**Discussion by tranb3r** [getsentry/sentry-dotnet#3397] [Q&A] [answered]
I'm using a logger to send errors to sentry:
`Logger.LogError("Error -- foo={CompanyFoo}", 42);`
This works and creates an error with a tag named `CompanyFoo`.
And in the console I can see the message `Error -- foo=42` which is what I expect.

Now, what if I want to have a tag named `Company.Foo` (with a dot)?
Let's try `Logger.LogError("Error -- foo={Company.Foo}", 42);`
This works for Sentry: the tag is created correctly.
However, `Company.Foo` is not a valid property name for the logger. It should be a valid C# variable name, so no dot allowed.
As a result, in the console, I get this message `Error -- foo={Company.Foo}` instead of `Error -- foo=42`.

Is there a way to fix this?

**Comment by bitsandfoxes** [comment]
Hey @tranb3r, could you provide a little demo snippet of what it is you're doing? So I can see what integrations are being used and to dig into this?

**Reply by tranb3r** [reply]
Sure. Here is a repro:  https://github.com/tranb3r/Issues/tree/main/ConsoleAppSentryTag
Just make sure you insert your own Dsn.

So, I'm doing two tests in this console app: first test is using a default logger, second test is using Serilog. Both loggers are writing to the console and sending the error to Sentry.
As you can see, when using a dot in a tag name, the console log is ok when using the default logger, but it's not ok when using Serilog.
In both cases, the Sentry issue is ok (the tag is correctly parsed, and the dot is correctly interpreted).

So I guess, this is a more a Serilog issue. 
And my question is, is there a workaround that could allow using a tag with a dot, when using both Serilog and Sentry.
Like, maybe there is another character (instead of dot) we can use, that is accepted by Serilog, and interpreted by Sentry as a dot?

**Comment by tranb3r** [comment, marked as answer]
I've just realized a fix is actually in the roadmap for Serilog v4.
https://github.com/serilog/serilog/pull/2063
So I guess we'll just have to wait for it.

**Reply by bitsandfoxes** [reply]
Thanks for the update!
